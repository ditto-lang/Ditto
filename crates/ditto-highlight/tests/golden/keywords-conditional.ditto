module Test exports (..);


bool = fn (b, x, y) -> if b then x else y;

unwrap = fn (wrapped) ->
    match wrapped with
    | Wrapped(x) -> x
    end;
