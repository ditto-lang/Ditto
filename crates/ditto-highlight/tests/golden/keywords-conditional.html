<KeywordImport>module</KeywordImport> <Namespace>Test</Namespace> <KeywordImport>exports</KeywordImport> <Bracket>(</Bracket><Symbol>..</Symbol><Bracket>)</Bracket><Delimiter>;</Delimiter>


<TopLevelName>bool</TopLevelName> = <Keyword>fn</Keyword> <Bracket>(</Bracket><Variable>b</Variable><Delimiter>,</Delimiter> <Variable>x</Variable><Delimiter>,</Delimiter> <Variable>y</Variable><Bracket>)</Bracket> <Operator>-></Operator> <KeywordConditional>if</KeywordConditional> <Variable>b</Variable> <KeywordConditional>then</KeywordConditional> <Variable>x</Variable> <KeywordConditional>else</KeywordConditional> <Variable>y</Variable><Delimiter>;</Delimiter>

<TopLevelName>unwrap</TopLevelName> = <Keyword>fn</Keyword> <Bracket>(</Bracket><Variable>wrapped</Variable><Bracket>)</Bracket> <Operator>-></Operator>
    <KeywordConditional>match</KeywordConditional> <Variable>wrapped</Variable> <KeywordConditional>with</KeywordConditional>
    <Delimiter>|</Delimiter> <EnumMember>Wrapped</EnumMember><Bracket>(</Bracket><Variable>x</Variable><Bracket>)</Bracket> <Operator>-></Operator> <Variable>x</Variable>
    <KeywordConditional>end</KeywordConditional><Delimiter>;</Delimiter>
